src/main/java/org/zalando/nakadi/validation/SchemaEvolutionService.java

```diff
@@ -97,6 +102,12 @@ public EventType evolve(final EventType original, final EventTypeBase eventType)
 
         final Version.Level changeLevel = semanticOfChange(changes, original.getCompatibilityMode());
 
+        if (changeLevel == NO_CHANGES && !original.getSchema().getSchema().equals(eventType.getSchema().getSchema())) {
+            LOG.error("undetected schema changes from {} to {}", original.getSchema().getSchema(),
+                    eventType.getSchema().getSchema());
+            throw new MyNakadiRuntimeException1("undetected schema changes");
```

Comment:
@rcillo I don't really see how it is connected. When we throw an exception we should try it to be as informative as possible so that the one who catches it will make the most use of it. Throwing default MyNakadiRuntimeException1 just makes it very uninformative like just throwing a plain RuntimeException.

Also, I'm not sure that it is completely unexpected. At least in the ticket I see a description of the case when this can happen. https://techjira.zalando.net/browse/ARUHA-677